### PR TITLE
Report migration policies as part of the k8s reporter when tests finish

### PR DESCRIPTION
**Pull request by iholder101**
### What this PR does
Before this PR:
Migration policies are not dumped as part of the `AfterSuite` after tests finish.

After this PR:
Migration policies are dumped as part of the `AfterSuite` after tests finish.
### Additional context
Tested locally:
```shell
> cat _out/artifacts/k8s-reporter/1_migrationpolicies.log 
{
    "metadata": {
        "resourceVersion": "247717"
    },
    "items": [
        {
            "kind": "MigrationPolicy",
            "apiVersion": "migrations.kubevirt.io/v1alpha1",
            "metadata": {
                "name": "testpolicy-vnmz7",
                "uid": "fd372a97-7974-46a5-b58f-db7039329f26",
                "resourceVersion": "247708",
                "generation": 1,
                "creationTimestamp": "2024-09-16T07:58:39Z",
                "labels": {
                    "test.kubevirt.io/kubevirt-test-default1": ""
                }
            },
            "spec": {
                "selectors": {
                    "virtualMachineInstanceSelector": {
                        "testpolicy-vnmz7-key-0": "testpolicy-vnmz7-value-0"
                    }
                }
            },
            "status": {}
        }
    ]
}
```

(file is trimmed for simplicity).

This checklist is not enforcing, but it's a reminder of items that could be relevant to every PR.
Approvers are expected to review this list.

- [ ] ~Design: A [design document](https://github.com/kubevirt/community/tree/main/design-proposals) was considered and is present (link) or not required~
- [X] PR: The PR description is expressive enough and will help future contributors
- [X] Code: [Write code that humans can understand](https://en.wikiquote.org/wiki/Martin_Fowler#code-for-humans) and [Keep it simple](https://en.wikipedia.org/wiki/KISS_principle)
- [X] Refactor: You have [left the code cleaner than you found it (Boy Scout Rule)](https://learning.oreilly.com/library/view/97-things-every/9780596809515/ch08.html)
- [ ] ~Upgrade: Impact of this change on upgrade flows was considered and addressed if required~
- [X] Testing: New code requires [new unit tests](https://github.com/kubevirt/kubevirt/blob/main/docs/reviewer-guide.md#when-is-a-pr-good-enough). New features and bug fixes require at least on e2e test
- [ ] ~Documentation: A [user-guide update](https://github.com/kubevirt/user-guide/) was considered and is present (link) or not required. You want a user-guide update if it's a user facing feature / API change.~
- [ ] ~Community: Announcement to [kubevirt-dev](https://groups.google.com/g/kubevirt-dev/) was considered~

### Release note

```release-note
NONE
```

### CODE REVIEW

**Comment by iholder101** [comment]
@enp0s3 ready for review :) PTAL

File: tests/reporter/kubernetes.go [hunk 1]
```diff
@@ -144,14 +145,14 @@ func (r *KubernetesReporter) ReportSpec(specReport types.SpecReport) {
 }
 
 func (r *KubernetesReporter) DumpTestNamespaces(duration time.Duration) {
-	r.dumpNamespaces(duration, testsuite.TestNamespaces)
+	r.dumpTestObjects(duration, testsuite.TestNamespaces)
 }
 
 func (r *KubernetesReporter) DumpAllNamespaces(duration time.Duration) {
```
Comment (Barakmor1) [inline tests/reporter/kubernetes.go:151]: Shouldn't we change these functions name as well?

**Comment by Barakmor1** [comment]
@iholder101 I can't see the artifacts in https://gcsweb.ci.kubevirt.io/gcs/kubevirt-prow/pr-logs/pull/kubevirt_kubevirt/12834/pull-kubevirt-e2e-k8s-1.31-sig-compute/1835595854719750144/artifacts/

File: tests/reporter/kubernetes.go (hunk 1 above)
Comment (iholder101) [reply tests/reporter/kubernetes.go]: Good point. Done!

**Comment by iholder101** [comment]
> @iholder101 I can't see the artifacts in [gcsweb.ci.kubevirt.io/gcs/kubevirt-prow/pr-logs/pull/kubevirt_kubevirt/12834/pull-kubevirt-e2e-k8s-1.31-sig-compute/1835595854719750144/artifacts](https://gcsweb.ci.kubevirt.io/gcs/kubevirt-prow/pr-logs/pull/kubevirt_kubevirt/12834/pull-kubevirt-e2e-k8s-1.31-sig-compute/1835595854719750144/artifacts/)

Because the test passed, so artifacts are not being saved.

**Review by enp0s3 — COMMENTED** [review]
/approve
@iholder101 Thank you!

**Pull request merged**